#### Every value has a type. In Python, the basic data types are numbers (ints and floats), booleans and strings.

In [4]:
#integer -> int
daysOfWeek = 7
#Float of decimal
Pi = 3.14159
#boolean -> bool
isMyFirstVariable = False
#String --> you can use single or double quotation
name = 'John'

#　you can print out the value of a variable by just typing its name
Pi, isMyFirstVariable, name

(3.14159, False, 'John')

In [5]:
#this code display what type is a varible using the function type (variable name)
type (daysOfWeek)

int

In [6]:
# suppose that the annual interest rate of a savings account is 4 %.
#To calculate the amount of money in the account after three years, 
# with an initialsum of 3, 000 dollars is put into the account
3000*1.04**3

3374.592

In [5]:
#  suppose that the area of a square is 10 m2. 
# The length of each edge can be calculated by
10 ** 0.5

3.1622776601683795

In [7]:
#  The result can be rounded up to the second decimal place.

round(10**0.5, 2)

3.16

In [10]:
#For notational convenience and to make programs easier to maintain
#Python allows names to be given to mathematical values. 
#An equivalent way of calculating the edge length is:
a=10
w=a**0.5
round(w,2)

3.16

In the example above, a denotes the area of the square, and w denotes its width.
The use of a and w makes it easier to understand the underlying physical meanings
of the values. a and w are called identifiers in Python. Each Python identifiers is
bound to a specific value. In the example, a is bound to 10 and w is bound to √10.


In [11]:
#Identifiers can be bound to new value:
x=1
x

1

In [12]:
x=2
x

2


In the example, the value of x is first 1, and then 2. Because identifiers can change
their values, they are also called variables.

The = sign in the above example is not an operator, and hence the commands a = 10 and w = round(a ∗ ∗0.5) are not expressions. They bare no values. Instead,the = sign denotes an assignment statement, which binds an identifier to a value.
Here a statement is a command to be executed by Python, and statements are the basic execution units in Python.
Python gives a name to a value by binding the value to an identifier.

Formally, an identifier must start with a letter or underscore (_), and contain a sequence of letters, numbers and underscores. 
For example, area, a, a0, area_of_square and _a are all valid identifiers, but 0a, area of square or a1! are not valid identifiers. An additional rule is that identifiers must not be keywords in Python, which are a list of reserved words.

In [14]:
#suppose that a ball is tossed up on the edge of a cliff with an initial velocity v0, 
#and that the initial altitude of the ball is 0 m. 
#The question is to find the vertical position 
# of the ball at a certain number of seconds t after the toss. 
#If the initial velocity of 5 m/s and the time is 0.1 s, the altitude can be calculated by
v0=5
g=9.81
t=0.1
h=v0*t-0.5*g*t**2
round(h,2)

0.45

In [15]:
#to further obtain the vertical location of the ball at 1 s, 
#only t and h need to be modified.
t=1
h=v0*t-0.5*g*t**2
round(h,2)

0.09

In [16]:
# Note that the value of h must be calculated again after the value of t changes.
#This is because an assignment statement binds an identifier to a value, 
#rather than establishing a mathematical correlation between a set of variables.
x=1
x=x+1
x

2

In [18]:
#An equivalent but perhaps less ‘counter-intuitive’ way of doing x = x + 1 is x+= 1.

x=1
x+=1
x

2

In [19]:
#The same applies to x = x − 3, x = x ∗ 6, and other arithmetic operators
x-=3
x


-1

In [20]:
x*=6
x

-6

In general, x<op>=y is equivalent to x=x<op>y, where <op> can be +, −, ∗,
/, % etc. The special assignment statements +=, −=, ∗=, /= and %= can be used
as a concise alternative to a = assignment statement when it incrementally changes
the value of one variable.

In [21]:
#to find the position of the ball after 3 s in the previous problem.

t=3
round(h,2)

0.09

In [23]:
h=v0*t-0.5*g*t**2
round(h,2)

-29.15

The commands above are executed sequentially and individually. 
When t is bound to the new value 3, h is not affected, and remains 0.09. 
After the new h assignment is executed, its value changes to −29.15 according to the new t.

In [25]:
#Cascaded assignments.
#Several assignment statements to the same value can be cascaded into a single line.
#For example, a = 1 and b = 1 can be cascaded into a = b = 1.
a=b=1
a


1

In [26]:
b

1

In [33]:
# float/ decimal
Pi=3.14159

#boolean ->bool
isMyFirstVatiable = False

#string(single/double quotation)
name='EE'

#Display the types
type(Pi), type(isMyFirstVatiable), type(name)






(float, bool, str)

In [37]:
# Integer
age=3

#Float
weight=200

#Bool
employed = True

# String

name='KK'

age, type(age),weight,type(age),employed,type(employed),name,type(name)

(3, int, 200, int, True, bool, 'KK', str)